yt_dlp/extractor/youtube.py

```diff
@@ -2452,11 +2469,11 @@ def _is_agegated(player_response):
     def _is_unplayable(player_response):
         return traverse_obj(player_response, ('playabilityStatus', 'status')) == 'UNPLAYABLE'
 
-    def _extract_player_response(self, client, video_id, master_ytcfg, player_ytcfg, identity_token, player_url, initial_pr):
+    def _extract_player_response(self, client, video_id, master_ytcfg, player_ytcfg, identity_token, player_url, initial_pr, require_js_player):
 
         session_index = self._extract_session_index(player_ytcfg, master_ytcfg)
         syncid = self._extract_account_syncid(player_ytcfg, master_ytcfg, initial_pr)
-        sts = self._extract_signature_timestamp(video_id, player_url, master_ytcfg, fatal=False)
+        sts = self._extract_signature_timestamp(video_id, player_url, master_ytcfg, fatal=False) if require_js_player else None
```

Comment:
Is there really a need for `require_js_player`? Can't we just check if `player_url is None`? I also noticed `fatal=true` is never used for `_extract_signature_timestamp`

Reply:
require js player: for the case where we have a player url but the particular client does not require it, from memory. Though thinking now we could remove that parameter and just do `player_url=player_url if require_js_player else None` on the func call.

I think the fatal thing was just to keep consistency with the signature extraction function originally. Could probably be set to be not fatal by default imo.

Reply:
> require js player: for the case where we have a player url but the particular client does not require it, from memory. Though thinking now we could remove that parameter and just do player_url=player_url if require_js_player else None on the func call.

Unless I am mistaken, the below code ensures that `player_url` is never set if not `require_js_player` 

> I think the fatal thing was just to keep consistency with the signature extraction function originally. Could probably be set to be not fatal by default imo

I meant that the function is never called with `fatal=True` anywhere in the code. So I was thinking that whole parameter can be scraped. But if you think it will be usefull in future, let it be

Reply:
> Unless I am mistaken, the below code ensures that player_url is never set if not require_js_player

nah. `player_url` is only cleared when the player_skip arg is given. We don't clear it otherwise since we might have other clients in the iteration that may need it. `require_js_player` just tells us if we need to worry about having a player (and sts) for the current client in the iteration.